architect.yml

```diff
@@ -19,43 +21,43 @@ secrets:
     default: secret
   db_name:
     description: Name of the database the component will store content in
-    default: api-db
+    default: app-db
   db_port:
     description: Port for the database
     default: 5432
-  api_port:
-    description: Port for api
+  app_port:
+    description: Port the app accepts connections on
     default: 8080
 
 services:
   ### This service is the actual Node.js component ###
-  api:
+  app:
     # Build context is the location of your application relative to the architect.yml file
     build:
       context: .
     depends_on:
       - database
     interfaces:
       main:
-        port: ${{ secrets.api_port }}
+        port: ${{ secrets.app_port }}
         ingress:
-          subdomain: api
+          subdomain: app
     # The liveness_probe makes sure the application is healthy - if not, it will re-start the application
     liveness_probe:
-      command: curl --fail 0.0.0.0:8080
+      command: curl --fail localhost:8080
       interval: 5s
       failure_threshold: 3
     # Set environment variables for your application here
     environment:
       DB_ADDR: ${{ services.database.interfaces.main.url }}/${{ secrets.db_name }}
       DB_USER: ${{ secrets.db_user }}
       DB_PASS: ${{ secrets.db_pass }}
-      PORT: ${{ secrets.api_port }}
+      PORT: ${{ secrets.app_port }}
     # Debug block is used to enable hot-reloading to make real-time changes to an
     #   application while running `architect dev`. For more info, see -
     #   https://docs.architect.io/components/local-configuration/#hot-reloading
     debug:
-      command: npm run dev
+      command: sh -c 'npm install && npm run dev'

```

Comment:
Why does this need to run `npm install` here? The Dockerfile already does this. https://github.com/architect-templates/nodejs/blob/main/Dockerfile#L8

Reply:
Glad you asked - I might be missing something but this is the only solution I could come up with 😂 . The reason is because we mount `.` which seems to wipe the `node_modules` that got installed in the container. If you remove this `npm install` and run `architect dev` with no `node_modules` locally, it errors despite the install in the Dockerfile

Reply:
Why are we mounting `.`? We should be mounting just `src`

```yml
volumes:
        src:
          mount_path: /usr/src/app/src
          host_path: ./src
```

Reply:
Yep that sounds right, will make that change lol